### PR TITLE
[SD-324] Jira test notification

### PR DESCRIPTION
**Pull request by klaukl**
Runs when on uat or master branch and - use commit sha from the triggered commit to get jira ID i.e. SCFA-123. May have edge case issues but any updates we want reflected via release probably should have a Jira ID otherwise it will be out of sync

https://digital-vic.atlassian.net/browse/SD-324

Implementation test on reference: https://github.com/dpc-sdp/reference-sdp-vic-gov-au/actions/runs/11472522262
```
  notify_jira:
    needs: [fe_smoke, fe_core_visual, fe_custom_visual, fe_custom, fe_core]
    if: (github.ref_name == 'uat' || github.ref_name == 'master')
    uses: dpc-sdp/github-actions/.github/workflows/notify_jira.yml@main
    secrets: inherit
    with:
      status: ${{ 
        (needs.fe_smoke.result == 'failure' || 
        needs.fe_core_visual.result == 'failure' || 
        needs.fe_custom_visual.result == 'failure' || 
        needs.fe_custom.result == 'failure' || 
        needs.fe_core.result == 'failure') 
        && 'failed' || 'passed' }}
```

### CODE REVIEW

File: .github/workflows/post_jira_comment.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        default: ""
+
+jobs:
+  post_jira_comment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Post JIRA Comment
+        id: post_jira_comment
+        env:
+          JIRA_USERNAME: ${{ secrets.JIRA_USERNAME }}
```
Comment (klaukl) [inline .github/workflows/post_jira_comment.yml:22]: Should we use a TEST_JIRA_USERNAME secret to abstract out stuff used to testing?

File: .github/workflows/test_prepare_jira.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+          # Construct the comment template
+          COMMENT="E2E Test - ${STATUS} on branch ${GIT_BRANCH}\n"
+          COMMENT+="Workflow URL: $WORKFLOW_LINK"
+
+          # Print the comment to the console for debugging
+          echo -e "COMMENT:\n${COMMENT}"
+          echo "COMMENT=$COMMENT" >> $GITHUB_ENV
+          echo "comment=$COMMENT" >> "$GITHUB_OUTPUT"
+
+  post_jira_comment:
```
Comment (klaukl) [inline .github/workflows/test_prepare_jira.yml:75]: Moved it out as requested @GROwen

File: .github/workflows/test_prepare_jira.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
+name: test_prepare_jira
```
[the remark this answers is not in the document]
Comment (klaukl) [reply .github/workflows/test_prepare_jira.yml:1]: The one below is already post_jira_comment though

[the remark this answers is not in the document]
Comment (tim-yao) [reply .github/workflows/test_prepare_jira.yml:1]: Does it matter?

**Comment by tim-yao** [comment]
Hey, @GROwen Keith have done a code split to make the workflow more reusable, could you please review it to see if it addresses your suggestion? Cheers!

File: .github/workflows/test_prepare_jira.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (klaukl) [reply .github/workflows/test_prepare_jira.yml]: Changed. Done

File: .github/workflows/test_post_jira.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            echo "jira_id=$JIRA_ID" >> "$GITHUB_OUTPUT"
+
+          fi
+  
+      - name: Set the workflow status
+        id: set_e2e_status
+        if: env.JIRA_ID != ''
+        run: |
+          echo 'STATUS=${{ inputs.status }}' >> $GITHUB_ENV
+          echo "Overall workflow status: ${STATUS}"
```
Comment (tim-yao) [inline .github/workflows/test_post_jira.yml:53]: Do we need this step?

File: .github/workflows/test_post_jira.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        env:
+          WORKFLOW_LINK: "https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}"  # Link to the workflow run
+        run: |
+          echo "Preparing JIRA comment..."
+
+          # Determine the branch name from the Git reference
+          GIT_BRANCH=${GITHUB_REF##*/}
+
+          # Construct the comment template
+          COMMENT="E2E Test - ${STATUS} on branch ${GIT_BRANCH}\n"
```
Comment (tim-yao) [inline .github/workflows/test_post_jira.yml:67]:
```suggestion
          COMMENT="E2E Test - ${inputs.status} on branch ${GIT_BRANCH}\n"
```

File: .github/workflows/test_post_jira.yml [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+          COMMENT+="Workflow URL: $WORKFLOW_LINK"
+
+          # Print the comment to the console for debugging
+          echo -e "COMMENT:\n${COMMENT}"
+          echo "COMMENT=$COMMENT" >> $GITHUB_ENV
```
Comment (tim-yao) [inline .github/workflows/test_post_jira.yml:72]: Do we need this env var?

Comment (klaukl) [reply .github/workflows/test_post_jira.yml:65]: Removed the whole section

File: .github/workflows/test_post_jira.yml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+          JIRA_ID=$(echo "$commit_message" | grep -oE 'SCFA-[0-9]{1,5}' || true)  # Prevents non-zero exit code
+  
+          # Check if JIRA ID is found
+          if [ -z "$JIRA_ID" ]; then
+            echo "No JIRA ID found in commit message: $commit_message"
+          else
+            echo "Extracted JIRA ID: $JIRA_ID"
+            echo "JIRA_ID=$JIRA_ID" >> $GITHUB_ENV
+            echo "jira_id=$JIRA_ID" >> "$GITHUB_OUTPUT"
+
```
Comment (tim-yao) [inline .github/workflows/test_post_jira.yml:45]: Empty line

**Review by tim-yao — APPROVED** [review]

File: .github/workflows/test_prepare_jira.yml (hunk 2 above)
Comment (klaukl) [reply .github/workflows/test_prepare_jira.yml]: Hey @GROwen any chances to get a check? :) Thanks

File: .github/workflows/post_jira_comment.yml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+          JIRA_PASSWORD: ${{ secrets.JIRA_PASSWORD }}
+        run: |
+          echo "Received JIRA ID: ${{ inputs.jira_id }}"
+          echo "Received COMMENT: ${{ inputs.comment }}"
+          echo "JIRA_ID=${{ inputs.jira_id }}" >> $GITHUB_ENV
+          echo "COMMENT=${{ inputs.comment }}" >> $GITHUB_ENV
```
Comment (GROwen) [inline .github/workflows/post_jira_comment.yml:28]: These shouldn't be necessary. Not when using [inputs.jira_id](https://github.com/dpc-sdp/github-actions/pull/56/files#diff-f36580701d0bab22dfffbcd283525ee70e0ffc392ab23ed71f558debafefa35eR31) and [inputs.comment](https://github.com/dpc-sdp/github-actions/pull/56/files#diff-f36580701d0bab22dfffbcd283525ee70e0ffc392ab23ed71f558debafefa35eR37) which is preferred 👍

File: .github/workflows/post_jira_comment.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            "${JIRA_API_URL}")
+
+          http_code="${response:(-3)}"  # Extract HTTP status code (last 3 characters)
+          response_body="${response:0:-3}"  # The response body
+
+          # Handle the API response
+          if [ "$http_code" -eq 201 ]; then
+            echo "Comment posted successfully to Jira issue ${{ inputs.jira_id }}"
+          else
+            echo "Failed to post comment to Jira issue ${{ inputs.jira_id }}. HTTP Code: $http_code"
```
Comment (GROwen) [inline .github/workflows/post_jira_comment.yml:47]: Will this fail the workflow? I don't think it should but it would be good to flag the failure in a more obvious way. Maybe by [updating the jobs summary](https://github.blog/news-insights/product-news/supercharging-github-actions-with-job-summaries/)?

**Review by GROwen — CHANGES_REQUESTED** [review]

File: .github/workflows/post_jira_comment.yml (hunk 9 above)
Comment (klaukl) [reply .github/workflows/post_jira_comment.yml:45]: It won't fail the workflow - want to mass test it better when we have pushed and use it before disrupting the whole workflow.

I've added the summary message

image

File: .github/workflows/post_jira_comment.yml (hunk 8 above)
Comment (klaukl) [reply .github/workflows/post_jira_comment.yml]: Removed

**Comment by klaukl** [comment]
@GROwen changed requested done. Could you please check? Thanks

**Review by GROwen — APPROVED** [review]
